File: solo_explore/lib/models/reward.dart
class Reward {
  final int id;
  final String name;
  final String slug;
  final String description;
  final String? image;
  final int requiredPoints;
  final String type;
  final String? value;
  final String? validUntil;
  final bool isActive;
  final bool isClaimed;
  final String? claimedAt;
  final String? code;
  final String? status;

  Reward({
    required this.id,
    required this.name,
    required this.slug,
    required this.description,
    this.image,
    required this.requiredPoints,
    required this.type,
    this.value,
    this.validUntil,
    this.isActive = true,
    this.isClaimed = false,
    this.claimedAt,
    this.code,
    this.status,
  });

  factory Reward.fromJson(Map<String, dynamic> json) {
    return Reward(
      id: json['id'],
      name: json['name'],
      slug: json['slug'],
      description: json['description'] ?? '',
      image: json['image'],
      requiredPoints: json['required_points'] ?? 0,
      type: json['type'] ?? '',
      value: json['value'],
      validUntil: json['valid_until'],
      isActive: json['is_active'] ?? true,
      isClaimed: json['is_claimed'] ?? false,
      claimedAt: json['claimed_at'],
      code: json['code'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'slug': slug,
      'description': description,
      'image': image,
      'required_points': requiredPoints,
      'type': type,
      'value': value,
      'valid_until': validUntil,
      'is_active': isActive,
      'is_claimed': isClaimed,
      'claimed_at': claimedAt,
      'code': code,
      'status': status,
    };
  }
}

File: Solo_Explore-fitur-umar/solo-explore-backend/app/Models/Reward.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Reward extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'slug',
        'description',
        'image',
        'required_points',
        'type',
        'value',
        'valid_until',
        'is_active',
    ];

    protected $casts = [
        'required_points' => 'integer',
        'valid_until' => 'date',
        'is_active' => 'boolean',
    ];

    // Relationships
    public function users()
    {
        return $this->belongsToMany(User::class, 'user_rewards')
            ->withTimestamps()
            ->withPivot(['claimed_at', 'used_at', 'status', 'code']);
    }

    // Scopes
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    public function scopeAvailableFor($query, $userPoints)
    {
        return $query->where('required_points', '<=', $userPoints);
    }
}
